common/kvstore/KeyValueStore.cc

```diff
@@ -57,7 +63,9 @@ KeyValueStore::KeyValueStore(string version, string path, string flavor)
     if (rc != 0) {
         goto fail;
     }
-    rc = mdb_env_open(dbState->env, this->path.c_str(), 0, 0664);
+    // _disable_ thread local storage so that any thread can close/abort a reader transaction.
```

Comment:
what implication does disabling it have? Let's document it

Reply:
Will do.

This option stores transactions in the transaction objects directly rather than thread-local storage, enabling read transactions to migrate threads. Users have to enforce that they are not used concurrently by multiple threads (which we do). We require this migration to allow the writer thread to close reader transaction on kvstore destruction.